core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManager.java

```diff
@@ -270,19 +270,23 @@ public InputStream fetchLogSegment(final RemoteLogSegmentMetadata remoteLogSegme
     public InputStream fetchLogSegment(final RemoteLogSegmentMetadata remoteLogSegmentMetadata,
                                        final int startPosition,
                                        final int endPosition) throws RemoteStorageException {
-        metrics.recordSegmentFetch();
+        final long startedMs = time.milliseconds();
 
         try {
-            final SegmentManifest segmentManifest = segmentManifestProvider.get(remoteLogSegmentMetadata);
-
             final BytesRange range = BytesRange.of(
                 startPosition,
                 Math.min(endPosition, remoteLogSegmentMetadata.segmentSizeInBytes() - 1)
             );
+            metrics.recordSegmentFetch(range.size());
```

Comment:
This will be skewed almost always: we request from `startPosition` to the end of file, but read only so many bytes on the broker side. We probably need this metric inside some wrapper `InputStream` that counts read bytes and register on close.

Reply:
Agree, though I think we need both. We can have this as the broker-side request (hopefully with proper ranges in the short future), and at the input stream level as you suggest to study both metrics, wdyt?

Reply:
If we want to keep this metric here I would suggest to rename it to something like `segment-fetch-requested` or `segment-fetch-bytes-requested` or whatever else saying that this is the amount of bytes that was requested by broker but neither the amount of bytes that was in the end fetched from remote storage nor the amount of bytes that was actually read by the broker.